server/datastore/mysql/hosts.go

```diff
@@ -1615,7 +1615,11 @@ WHERE
 	}
 
 	if err := sqlx.GetContext(ctx, ds.reader, &row, query, args...); err != nil {
-		return nil, err
+		if err == sql.ErrNoRows {
+			return nil, ctxerr.Wrap(ctx, notFound("OSVersions"))
+		} else {
+			return nil, err
+		}
```

Comment:
Nit: see the lint warning above. Whenever it's possible to remove indentation, we should do so for readability.